crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -110,20 +114,42 @@ impl ParsedQProperty {
                 let mut write = None;
                 let mut notify = None;
 
+                let mut constant = false;
+                let mut required = false;
+                let mut is_final = false;
+                let mut reset = None;
+
+                let map_auto_or_custom = |variable: &mut Option<FlagState>, value: &Option<Ident>| {
+                    *variable = Some(value.as_ref().map_or(FlagState::Auto, |ident| FlagState::Custom(ident.clone())));
+                };
+
                 // Create mutable closure to capture the variables for setting with the Meta values
                 let mut update_fields = |ident: &Ident, value: Option<Ident>| -> Result<()> {
-                    let variable = match ident.to_string().as_str() {
-                        "read" => &mut read,
-                        "write" => &mut write,
-                        "notify" => &mut notify,
+                    match ident.to_string().as_str() {
+                        "READ" => map_auto_or_custom(&mut read, &value),
+                        "WRITE" => map_auto_or_custom(&mut write, &value),
+                        "NOTIFY" => map_auto_or_custom(&mut notify, &value),
+                        "CONSTANT" => constant = true,
+                        "REQUIRED" => required = true,
+                        "FINAL" => is_final = true,
+                        "RESET" => {
+                            // Might be refactorable to be simpler
+                            if let Some(ident) = value {
+                                reset = Some(ident);
+                            } else {
+                                return Err(Error::new(
+                                    ident.span(),
+                                    "The identifier of a reset function must be provided explicitly",
+                                ));
+                            }
```

Comment:
that would create a Some(Ok(Ident)) so not that I don't think

Reply:
yes it would create `Some(OK(Ident))` initially but there is then the `?` at the end of the `ok_or_else` which unwraps it to either return an error or become `Some(Ident)`

Reply:
done